backend/app/api/auth.py

```
@@ -75,15 +79,66 @@ async def oauth_client_metadata(request: Request) -> dict:
+# better than "OAuth sign-in failed." (it already renders `OAuth failed: <reason>`).
+# Every value is a LITERAL authored here — never a provider-supplied string — so the
+# no-remote-input property of the redirect is preserved; the detail stays in the log.
+# Each code names only what we can actually tell apart, so none of them sends the
+# operator after the wrong thing:
+_REASON_PROVIDER_DENIED = "provider_denied"  # the AS said no (access_denied)
+_REASON_PROVIDER_UNAVAILABLE = "provider_unavailable"  # the AS is down/overloaded — retry
+_REASON_PROVIDER_REJECTED = "provider_rejected_request"  # the AS refused OUR request shape
+_REASON_NO_CODE = "no_code"  # redirect carried no code/state
+_REASON_EXPIRED = "expired_or_superseded"  # unknown state: reaped, cancelled, restarted
+_REASON_EXCHANGE_FAILED = "token_exchange_failed"  # the AS refused the code (SDK OAuthTokenError)
+_REASON_TOKEN_REFUSED = "upstream_refused_token"  # got a token; the RESOURCE rejected it (scopes)
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "_REASON_TOKEN_REFUSED"

(S105)